### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/detect-secrets/commit/92120a66fa6e90777ac53ae1b89e79274004dafa#commitcomment-77131949

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/detect-secrets/commit/2705dfe7ccb54fcd6436543c3331a467e88f24d3

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -211,7 +211,7 @@ the ["User Facing Changes"](https://github.com/Yelp/detect-secrets/pull/355) for
 #### :mortar_board: Walkthrough / Help
 
 - The README now includes examples of common usages, features, and an FAQ section for
-  the common questions we often receive as Github issues.
+  the common questions we often receive as GitHub issues.
```
Comment (jsoref) [inline CHANGELOG.md:214]: brand

File: detect_secrets/audit/compare.py [hunk 2]
```diff
@@ -222,7 +222,7 @@ def _display_difference_to_user(
             )
 
         if decision == io.InputOptions.QUIT:
-            io.print_message('Qutting...')
+            io.print_message('Quitting...')
```
Comment (jsoref) [inline detect_secrets/audit/compare.py:225]: user facing?

File: detect_secrets/plugins/base.py [hunk 3]
```diff
@@ -169,18 +169,18 @@ def build_assignment_regex(
         opt_quote = r'(?:"|\'|)'
         opt_open_square_bracket = r'(?:\[|)'
         opt_close_square_bracket = r'(?:\]|)'
-        opt_dash_undrscr = r'(?:_|-|)'
+        opt_dash_underscore = r'(?:_|-|)'
```
Comment (jsoref) [inline detect_secrets/plugins/base.py:172]: Happy to drop (this or anything else). Notably, in principle, this should really be `opt_underscore_dash`, but...

Note: Google Sheets made this suggestion (and it surprised me). I wouldn't have.

File: detect_secrets/settings.py [hunk 4]
```diff
@@ -221,7 +221,7 @@ def json(self) -> Dict[str, Any]:
         for plugin in get_plugins():
             # NOTE: We use the initialized plugin's JSON representation (rather than using
             # the configured settings) to deal with cases where plugins define their own
-            # default variables, that is not necessarily carried through through the
+            # default variables, that is not necessarily carried through the
```
Comment (jsoref) [inline detect_secrets/settings.py:224]: This is a fairly new heuristic, but I think it's right.

**Review by jsoref — COMMENTED** [review]
Most corrections were suggested by Google Sheets. All fault is mine.

File: CHANGELOG.md (hunk 1 above)
Comment (jpdakran) [reply CHANGELOG.md:214]: Nice!

File: detect_secrets/audit/compare.py (hunk 2 above)
Comment (jpdakran) [reply detect_secrets/audit/compare.py:225]: Nice!

File: detect_secrets/plugins/base.py (hunk 3 above)
Comment (jpdakran) [reply detect_secrets/plugins/base.py:172]: I'm okay with this.

File: detect_secrets/settings.py (hunk 4 above)
Comment (jpdakran) [reply detect_secrets/settings.py:224]: LGTM

**Review by jpdakran — APPROVED** [review]
This is great! All LGTM 👍

**Pull request merged**